### ISSUE TITLE
Invoking `substring` with max int spams server console with errors

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13101]
When running the following query:
```cypher
CREATE ({n0:(substring(" ", 1, 2147483647))}) CREATE ()
```

The server console fills up quickly with exactly 101 identical errors, their error message being just "null":
```
2023-03-30 12:43:22.894+0000 ERROR Client triggered an unexpected error [Neo.DatabaseError.Statement.ExecutionFailed]: null, reference 229386ff-a9b2-4a56-a08f-66f5ab32a5b2.
```

Client-side I get the following error:
```
TransactionExecutionLimit: Connection lost after 101 attempts, last error: expected success or database error
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.
Additionally, the server console's behaviour is unexpected.

I encountered this issue when testing queries against the **Neo4j 5.6.0 community version** in a Docker container running **alpine v.3**.
The client was using the [go-driver](https://github.com/neo4j/neo4j-go-driver).

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
CREATE ({n0:(substring(" ", 1, 2147483647))}) CREATE ()
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `TransactionExecutionLimit: Connection lost after 101 attempts, last error: expected success or database error`.
Additionally, the server console quickly fills up with 101 identical error messages.

**Comment by HannesSandberg** [comment]
Thank you! We will look into this.

**Comment by JonasHolgersson** [comment]
Thanks again for reporting, this has been fixed and should be included in the 5.8 release.
